cpp/src/strings/convert/convert_hex.cu

```diff
@@ -197,7 +194,8 @@ struct dispatch_integers_to_hex_fn {
   }
   // non-integral types throw an exception
   template <typename T, typename... Args>
-  std::enable_if_t<not std::is_integral_v<T>, std::unique_ptr<column>> operator()(Args...) const
+  std::enable_if_t<not std::is_integral_v<T> or std::is_same_v<T, bool>, std::unique_ptr<column>>
```

Comment:
nit: this seems like a common trait. consider adding it to traits.hpp

Reply:
There is an `cudf::is_index_type()` trait check that is equivalent to this but I felt the current code more accurately communicated the intention here.

Reply:
Then `is_integral_not_bool` should clearly communicate its intention.